File: Others/medium/BalancedBrackets.py
def balancedBrackets(string):

    brackets = {
        ')': '(',
        ']': '[',
        '}': '{'
    }

    stack = []

    for bracket in string:
        if bracket not in list(brackets.keys()) + list(brackets.values()):
            continue
        if bracket in brackets.values():
            stack.append(bracket)
        else:
            if not stack or brackets[bracket] != stack[-1]:
                return False
            stack.pop()
    return len(stack) == 0

File: Golang/medium/balanced-brackets.go
package main

type Stack []rune

func (s *Stack) Push(elem rune) {
	*s = append(*s, elem)
}

func (s *Stack) Pop() rune {
	elem := (*s)[len(*s)-1]
	*s = (*s)[:len(*s)-1]
	return elem
}

func (s *Stack) IsEmpty() bool {
	return len(*s) == 0
}

func BalancedBrackets(s string) bool {
	brackets := map[rune]rune{
		']': '[',
		')': '(',
		'}': '{',
	}

	allBrackets := []rune{'[', ']', '{', '}', '(', ')'}

	st := Stack{}

	for _, elem := range s {
		// I know I can have a set here but I am too lazy rn
		contains := false
		for _, br := range allBrackets {
			if br == elem {
				contains = true
				break
			}
		}
		if !contains {
			continue
		}
		val, closed := brackets[elem]
		if !closed {
			st.Push(elem)
		} else {
			if st.IsEmpty() {
				return false
			}
			top := st.Pop()
			if top != val {
				return false
			}
		}
	}

	return st.IsEmpty()
}
